### DISCUSSION TITLE
Questions regarding stream ciphers in advanced

### DISCUSSION THREAD

**Discussion by Chewhern** [jedisct1/libsodium#1241] [Q&A] [answered]
In the advanced of libsodium gitbook website, there's a sub tab of stream ciphers.

I can understand the functionality of xor and xor_ic but in every stream ciphers, there's this function called

```
int crypto_stream_chacha20(unsigned char *c, unsigned long long clen,
                           const unsigned char *n, const unsigned char *k); 
```

I couldn't quite get the functionality of this specific method/function. Can someone explain to me, what it does exactly?

**Comment by jedisct1** [comment, marked as answer]
It serves the same purpose as `randombytes_buf_deterministic()`: given a seed, it fills the output buffer with a deterministic sequence, that looks like random.

Unless you specifically need this function, you should just use `randombytes_buf_deterministic()` instead, if only because the name is self-explanatory.
